jira7/jira_data_mappings.bal

```diff
@@ -280,3 +259,164 @@ function jsonToIssueComment(json jcomment) returns IssueComment {
+
+function convertJsonToProjectCategory(json sourceGroupStruct) returns ProjectCategory|error {
+    ProjectCategory targetProjectCategory = {};
+    targetProjectCategory.resource_path = sourceGroupStruct.self != null ? sourceGroupStruct.self.toString() : "";
+    targetProjectCategory.id = sourceGroupStruct.id != null ? sourceGroupStruct.id.toString() : "";
+    targetProjectCategory.name = sourceGroupStruct.name != null ? sourceGroupStruct.name.toString() : "";
+    targetProjectCategory.description = sourceGroupStruct.description != null
+    ? sourceGroupStruct.description.toString() : "";
+    return targetProjectCategory;
+}
+
+function convertJsonToProject(json sourceGroupStruct) returns Project|error {
+    Project targetProject = {};
+    targetProject.resource_path = sourceGroupStruct.self != null ? sourceGroupStruct.self.toString() : "";
+    targetProject.id = sourceGroupStruct.id != null ? sourceGroupStruct.id.toString() : "";
+    targetProject.key = sourceGroupStruct.key != null ? sourceGroupStruct.key.toString() : "";
+    targetProject.name = sourceGroupStruct.name != null ? sourceGroupStruct.name.toString() : "";
+    targetProject.description = sourceGroupStruct.description != null ? sourceGroupStruct.description.toString() : "";
+    targetProject.leadName = sourceGroupStruct.leadName != null ? sourceGroupStruct.leadName.toString() : "";
+    targetProject.projectTypeKey = sourceGroupStruct.projectTypeKey != null
+    ? sourceGroupStruct.projectTypeKey.toString() : "";
+    targetProject.avatarUrls.^"48x48" = sourceGroupStruct.avatarUrls.^"48x48" != null
+    ? sourceGroupStruct.avatarUrls.^"48x48".toString() : "";
+    targetProject.avatarUrls.^"32x32" = sourceGroupStruct.avatarUrls.^"32x32" != null
+    ? sourceGroupStruct.avatarUrls.^"32x32".toString() : "";
+    targetProject.avatarUrls.^"24x24" = sourceGroupStruct.avatarUrls.^"24x24" != null
+    ? sourceGroupStruct.avatarUrls.^"24x24".toString() : "";
+    targetProject.avatarUrls.^"16x16" = sourceGroupStruct.avatarUrls.^"16x16" != null
+    ? sourceGroupStruct.avatarUrls.^"16x16".toString() : "";
+    targetProject.projectCategory.resource_path = sourceGroupStruct.projectCategory.self != null
+    ? sourceGroupStruct.projectCategory.self.toString() : "";
+    targetProject.projectCategory.id = sourceGroupStruct.projectCategory.id != null
+    ? sourceGroupStruct.projectCategory.id.toString() : "";
+    targetProject.projectCategory.name = sourceGroupStruct.self != null ? sourceGroupStruct.self.toString() : "";
+    targetProject.projectCategory.description = sourceGroupStruct.projectCategory.description != null
+    ? sourceGroupStruct.projectCategory.description.toString() : "";
+    targetProject.issueTypes = convertToIssueTypes(<json[]>sourceGroupStruct.issueTypes);
+    targetProject.components = convertToComponents(<json[]>sourceGroupStruct.components);
+    targetProject.versions = convertToVersions(<json[]>sourceGroupStruct.versions);
+    return targetProject;
```

Comment:
Can't we directly convert the json to Project? Check other places as well.
https://ballerina.io/learn/by-example/json-struct-map-conversion.html

Reply:
Yes,We can't do that. When I converted the json data to Project, I got the following error

incompatible stamp operation: 'json' value cannot be stamped as 'wso2/jira7:0.8.16:Project'